### PR TITLE
add component manila

### PR DESCRIPTION
**Pull request by Carthaca**
manila share service registers one service per enabled backend per host,
so we need to check all of them

### CODE REVIEW

File: agentliveness/agent.py [hunk 1]
```diff
@@ -145,6 +148,28 @@ def _check_cinder(self):
...
+        manila = manila_client.Client(session=self._get_session())
+        try:
+            for backend in self.CONF.enabled_share_backends:
+                if backend is not None:
+                    host = "{}@{}".format(self.CONF.host, backend)
+                else:
+                    host = self.CONF.host
+                for service in manila.services.list({'host': host}):
+                    if service.state == 'up':
+                        return 0
```
Comment (joker-at-work) [inline agentliveness/agent.py]: Do I understand this correctly: If the service for a single backend is up, the whole pod is fine? I see the point to restart the pod if one service is down, but not that a single service being up should determine the state of all services.

File: agentliveness/agent.py [hunk 2]
```diff
@@ -145,6 +148,28 @@ def _check_cinder(self):
...
+                    else:
+                        logger.error("Agent %s is down, commencing suicide", service.host)
+                        return 1
+
+                logger.warning("Agent hostname %s not registered" % host)
```
Comment (joker-at-work) [inline agentliveness/agent.py]: Shouldn't this be `, host)` instead of `% host)`?

File: agentliveness/agent.py (hunk 1 above)
Comment (Carthaca) [reply agentliveness/agent.py]: Thanks for finding. This is not intended, currently I always run only one backend per pod

File: agentliveness/agent.py (hunk 2 above)
Comment (Carthaca) [reply agentliveness/agent.py]: Yeah, copy&paste from cinder

**Review by joker-at-work — APPROVED** [review]

**Pull request merged**